Gravity/Gravity/DAL/RSAPI/RsapiDao.Insert.cs

```diff
@@ -29,55 +29,124 @@ protected int InsertRdo(RDO newRdo)
 
 		protected void InsertUpdateFileFields(BaseDto objectToInsert, int parentId)
 		{
-			foreach (var propertyInfo in objectToInsert.GetType()
-                .GetProperties()
-                .Where(c => c.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.File))
+			foreach (var propertyInfo in objectToInsert.GetType().GetProperties().Where(c => c.GetCustomAttribute<RelativityObjectFieldAttribute>()?.FieldType == RdoFieldType.File))
 			{
-                RelativityFile relativityFile = propertyInfo.GetValue(objectToInsert) as RelativityFile;
-                InsertUpdateFileField(relativityFile, parentId);
+				RelativityFile relativityFile = propertyInfo.GetValue(objectToInsert) as RelativityFile;
+				InsertUpdateFileField(relativityFile, parentId);
 			}
 		}
 
 		protected void InsertUpdateFileField(RelativityFile relativityFile, int parentId)
 		{
-            if (relativityFile?.FileValue == null)
-            {
-                return;
-            }
-
-            if (relativityFile.FileValue.Path != null)
-            {
-                rsapiProvider.UploadFile(relativityFile, parentId, relativityFile.FileValue.Path);
+			if (relativityFile?.FileValue == null)
+			{
+				return;
 			}
-            else if (!string.IsNullOrEmpty(relativityFile.FileMetadata.FileName))
-            {
-                string fileName = Path.GetTempPath() + relativityFile.FileMetadata.FileName;
-                File.WriteAllBytes(fileName, relativityFile.FileValue.Data);
-
-                try
-                {
-                    rsapiProvider.UploadFile(relativityFile, parentId, fileName);
+
+			if (relativityFile.FileValue.Path != null)
+			{
+				rsapiProvider.UploadFile(relativityFile, parentId, relativityFile.FileValue.Path);
+			}
+			else if (!string.IsNullOrEmpty(relativityFile.FileMetadata.FileName))
+			{
+				string fileName = Path.GetTempPath() + relativityFile.FileMetadata.FileName;
+				File.WriteAllBytes(fileName, relativityFile.FileValue.Data);
+
+				try
+				{
+					rsapiProvider.UploadFile(relativityFile, parentId, fileName);
+				}
+				finally
+				{
+					invokeWithRetryService.InvokeVoidMethodWithRetry(() => File.Delete(fileName));
 				}
-                finally
-                {
-                    invokeWithRetryService.InvokeVoidMethodWithRetry(() => File.Delete(fileName));
-                }
-                
-            }
-        }        
-
-        private void InsertChildListObjectsWithDynamicType(BaseDto theObjectToInsert, int resultArtifactId, PropertyInfo propertyInfo)
-        {
+			}
+		}
+
+		protected void InsertUpdateSingleObjectPropertiesBeforeRdoInsert(ref BaseDto parentObjectToInsert)
```

Comment:
It looks like there was some duplicate work done as there were already an open pull request that handled the single object insert.  Guess we should discuss and decide what implementation to go with.

Reply:
I suspect that this is similar to your first try on the other PR, where what we needed to do in the end was fix that `!= 0` statement.